client/component-lab/Notification/NotificationView.coffee

```diff
@@ -0,0 +1,236 @@
+      return
+
+
+    componentDidMount: ->
+
+      self = this
+      transitionEvent = whichTransitionEvent()
+      element = ReactDOM.findDOMNode(this)
+      notification = @props.notification
+      @_isMounted = true
+      if not @_noAnimation
+        if transitionEvent
+          element.addEventListener transitionEvent, @_onTransitionEnd
+        else
+          @_noAnimation = true
+      if not notification.dismissible and not notification.primaryButtonTitle and not notification.secondaryButtonTitle
+        @_notificationTimer = new (Helpers.Timer)((->
+          self._hideNotification()
+          return
+        ), notification.duration)
+      @_showNotification()
+      return
+
+
+    _handleMouseEnter: ->
+
+      notification = @props.notification
+      if not notification.dismissible and not notification.primaryButtonTitle and not notification.secondaryButtonTitle
+        @_notificationTimer.pause()
+      return
+
+
+    _handleMouseLeave: ->
+
+      notification = @props.notification
+      if not notification.dismissible and not notification.primaryButtonTitle and not notification.secondaryButtonTitle
+        @_notificationTimer.resume()
+      return
+
+
+    _handlePrimaryButtonClick: (event) ->
+
+      event.preventDefault()
+      notification = @props.notification
+      @_hideNotification()
+      if typeof notification.onPrimaryButtonClick is 'function'
+        notification.onPrimaryButtonClick()
+      return
+
+
+    _handleSecondaryButtonClick: (event) ->
+
+      event.preventDefault()
+      notification = @props.notification
+      @_hideNotification()
+      if typeof notification.onSecondaryButtonClick is 'function'
+        notification.onSecondaryButtonClick()
+      return
+
+
+    componentWillUnmount: ->
+
+      element = ReactDOM.findDOMNode(this)
+      transitionEvent = whichTransitionEvent()
+      element.removeEventListener transitionEvent, @_onTransitionEnd
+      @_isMounted = false
+      return
+
+
+    render: ->
+
+      notification = @props.notification
+      className = [
+        styles.kd_notification
+        styles["kd_notification_#{notification.type}"]
+      ]
+      iconClass = [
+        styles.kd_notification_icon
+        styles["kd_notification_icon_#{notification.type}"]
+      ]
+      closeButton = null
+      actions = null
+      actionsClass = null
+      primaryButton = null
+      secondaryButton = null
+      message = notification.content
+      if @state.visible
+        className.push styles.kd_notification_visible
+      else
+        className.push styles.kd_notification_hidden
+      if notification.dismissible
+        className.push styles.kd_notification_dismissible
+        closeButton = <button className={styles.kd_notification_close} onClick={this._dismiss}>&times;</button>
+      if notification.primaryButtonTitle
+        primaryButton = <div className={styles.kd_notification_action}>
+            <button type="button"
```

Comment:
Indentation!
For those kind of dom elements, you can separate them to out of class
[See this example](https://github.com/koding/koding/blob/master/client/home/lib/myteam/components/hometeamteammates/view.coffee#L63)